circuit/types/field/src/mul.rs

```diff
@@ -59,7 +59,7 @@ impl<E: Environment> MulAssign<Field<E>> for Field<E> {
 impl<E: Environment> MulAssign<&Field<E>> for Field<E> {
     fn mul_assign(&mut self, other: &Field<E>) {
         match (self.is_constant(), other.is_constant()) {
-            (true, true) | (false, true) => *self = (&self.linear_combination * *other.eject_value()).into(),
+            (true, true) | (false, true) => self.linear_combination *= &*other.eject_value(),
```

Comment:
This change allows us to go from `MulAssign<&Field<E>> for Field<E>` to the new `MulAssign<&F> for LinearCombination<F>` instead of going through `Mul<&F> for &LinearCombination<F>` (and ultimately `Mul<&F> for LinearCombination<F>`), which currently clones the left-hand side of this operation.